go/app/main.go

```diff
@@ -28,14 +39,63 @@ func root(c echo.Context) error {
 func addItem(c echo.Context) error {
 	// Get form data
 	name := c.FormValue("name")
+	category := c.FormValue("category")
 	c.Logger().Infof("Receive item: %s", name)
+	c.Logger().Infof("Receive item: %s", category)
 
 	message := fmt.Sprintf("item received: %s", name)
 	res := Response{Message: message}
 
+	// open json file & data
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		log.Fatal("JSONファイルを開けません", err)
+		return err
+	}
+	defer jsonFile.Close()
+
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		log.Fatal("JSONデータを読み込めません", err)
+		return err
+	}
+	// convert json into go format
+	var items Items
+	json.Unmarshal(jsonData, &items)
+	fmt.Println(items)
+	// add new item
+	newItem := Item{Name: name, Category: category}
+	items.Items = append(items.Items, newItem)
+
+	// convert go format into json
+	updatedJson, err := json.Marshal(&items)
+	if err != nil {
+		log.Fatal("JSONデータ変換に失敗", err)
+		return err
+	}
+	// output as json file
+	_ = ioutil.WriteFile("items.json", updatedJson, 0644)
```

Comment:
We shouldn't ignore the error here either. Checking errors is something we have to get used to in Go 😓 .